cpp/include/cudf/detail/utilities/device_operators.cuh

```diff
@@ -93,10 +117,17 @@ struct DeviceMax {
     return std::max(lhs, rhs);
   }
 
-  template <typename T>
+  template <typename T,
+            typename std::enable_if_t<!std::is_same<T, cudf::string_view>::value>* = nullptr>
   static constexpr T identity() {
     return std::numeric_limits<T>::lowest();
   }
+  template <typename T,
+            typename std::enable_if_t<std::is_same<T, cudf::string_view>::value>* = nullptr>
+  static constexpr T identity() {
+    return T{nullptr, 0};
```

Comment:
How do you handle identity for strings being nullptr for both min  and max?

Reply:
It's a hack at best. We really can't be using a null `string_view` as an identity for any operation. In fact, null `string_view`s should be removed entirely: https://github.com/rapidsai/cudf/issues/3346

Reply:
replaced `nullptr` with `__constant__ sentinel` device address. It's not used anyway because string_view `size_bytes() == 0`